integrationtests/src/test/java/io/kroxylicious/proxy/filter/ApiVersionsMarkingFilter.java

```diff
@@ -39,4 +39,22 @@ public CompletionStage<RequestFilterResult> onRequest(ApiKeys apiKey, RequestHea
             return context.forwardRequest(header, request);
         });
     }
+
+    public static class Factory implements FilterFactory<Void> {
```

Comment:
I realise that the generic is the config type but this reads very strangely to me as 
> This is a filter factory for Void things

Which immediately leads to the question why do we have a factory for void?

I do wonder if we should add the Filter type as a generic as well.

Reply:
yep, Tom was pointing that out as well that with a generic type we can make sure the filterType and the type returned by createFilter are consistent. So I've made that change.